File: src/main/java/com/github/kory33/signvote/manager/vote/VotePointSet.java
package com.github.kory33.signvote.manager.vote;

import com.github.kory33.signvote.vote.VotePoint;
import com.google.gson.JsonArray;
import com.google.gson.JsonPrimitive;

import java.util.HashSet;

/**
 * This class extends {@link HashSet} with a type parameter of {@link VotePoint}.
 *
 * No method is overrode, but a method {@link JsonArray} is added.
 */
/*package-private*/ class VotePointSet extends HashSet<VotePoint> {
    /**
     * Convert this object to a json-array that consists of strings representing name of each vote-point.
     * @return converted object
     */
    JsonArray toNameJsonArray() {
        JsonArray array = new JsonArray();

        this.stream()
                .map(VotePoint::getName)
                .map(JsonPrimitive::new)
                .forEach(array::add);

        return array;
    }
}

File: src/main/kotlin/com/github/kory33/signvote/manager/vote/VotePointSet.kt
package com.github.kory33.signvote.manager.vote

import com.github.kory33.signvote.vote.VotePoint
import com.google.gson.JsonArray
import com.google.gson.JsonPrimitive
import java.util.*

/**
 * This class extends [HashSet] with a type parameter of [VotePoint].

 * No method is overrode, but a method [JsonArray] is added.
 */
/*package-private*/ internal class VotePointSet : HashSet<VotePoint>() {
    /**
     * Convert this object to a json-array that consists of strings representing name of each vote-point.
     * @return converted object
     */
    fun toNameJsonArray(): JsonArray {
        val array = JsonArray()
        this.map({ JsonPrimitive(it.name) }).forEach({ array.add(it) })
        return array
    }
}
